pillar/base/haproxy.sls

```diff
@@ -94,6 +94,11 @@ haproxy:
       verify_host: salt.psf.io
       check: "GET /.well-known/acme-challenge/sentinel HTTP/1.1\\r\\nHost:\\ salt.psf.io"
 
+    publish-files:
+      domains: []
+      verify_host: salt-public.psf.io
```

Comment:
verify_host is for the backend hostname we'll verify the certificate against, so in this case it needs to match the certificate issued to the salt-master: `salt.psf.io`.